common/raw_hashtable.h

```diff
@@ -573,6 +573,7 @@ class BaseImpl {
   template <typename LookupKeyT>
   auto InsertIntoEmpty(LookupKeyT lookup_key, KeyContextT key_context)
       -> EntryT*;
```

Comment:
I suspect we can update the other caller to also pass in the hash, and only have the one signature?

Reply:
Removed this signature, updated callers. Also updated the signature of GrowAndInsert.

P. s.
analog of this function is absl is completely type erased. Not even part of the table.
https://github.com/abseil/abseil-cpp/blob/master/absl/container/internal/raw_hash_set.h#L1805-L1823

Not sure how relevant code size reduction is for Carbon. But that may make this function hotter and become iCache resident.